--- a/net/minecraft/entity/passive/EntitySheep.java
+++ b/net/minecraft/entity/passive/EntitySheep.java
@@ -1,25 +1,12 @@
 package net.minecraft.entity.passive;
 
 import com.google.common.collect.Maps;
-import java.util.Arrays;
-import java.util.Map;
-import java.util.Random;
-import java.util.stream.Collectors;
-import javax.annotation.Nullable;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.EntityAgeable;
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.SharedMonsterAttributes;
-import net.minecraft.entity.ai.EntityAIEatGrass;
-import net.minecraft.entity.ai.EntityAIFollowParent;
-import net.minecraft.entity.ai.EntityAILookIdle;
-import net.minecraft.entity.ai.EntityAIMate;
-import net.minecraft.entity.ai.EntityAIPanic;
-import net.minecraft.entity.ai.EntityAISwimming;
-import net.minecraft.entity.ai.EntityAITempt;
-import net.minecraft.entity.ai.EntityAIWanderAvoidWater;
-import net.minecraft.entity.ai.EntityAIWatchClosest;
+import net.minecraft.entity.ai.*;
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
@@ -36,12 +23,7 @@
 import net.minecraft.network.datasync.DataParameter;
 import net.minecraft.network.datasync.DataSerializers;
 import net.minecraft.network.datasync.EntityDataManager;
-import net.minecraft.util.DamageSource;
-import net.minecraft.util.EnumHand;
-import net.minecraft.util.IItemProvider;
-import net.minecraft.util.ResourceLocation;
-import net.minecraft.util.SoundEvent;
-import net.minecraft.util.Util;
+import net.minecraft.util.*;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.util.math.MathHelper;
 import net.minecraft.world.DifficultyInstance;
@@ -50,6 +32,12 @@
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+
 public class EntitySheep extends EntityAnimal
 {
     private static final DataParameter<Byte> DYE_COLOR = EntityDataManager.createKey(EntitySheep.class, DataSerializers.BYTE);
@@ -253,6 +241,9 @@
 
         if (itemstack.getItem() == Items.SHEARS && !this.getSheared() && !this.isChild())
         {
+            this.dropItems();
+            
+            /* [CM] Refactored code for dispenser sheep shearing
             if (!this.world.isRemote)
             {
                 this.setSheared(true);
@@ -270,13 +261,37 @@
                     }
                 }
             }
+            */
 
             itemstack.damageItem(1, player);
-            this.playSound(SoundEvents.ENTITY_SHEEP_SHEAR, 1.0F, 1.0F);
+            // this.playSound(SoundEvents.ENTITY_SHEEP_SHEAR, 1.0F, 1.0F); Refactored code for dispenser sheep shearing
         }
 
         return super.processInteract(player, hand);
     }
+    
+    // [CM] Moved code from processInteract
+    public void dropItems()
+    {
+        if (!this.world.isRemote)
+        {
+            this.setSheared(true);
+            int i = 1 + this.rand.nextInt(3);
+        
+            for (int j = 0; j < i; ++j)
+            {
+                EntityItem entityitem = this.entityDropItem(WOOL_BY_COLOR.get(this.getFleeceColor()), 1);
+            
+                if (entityitem != null)
+                {
+                    entityitem.motionY += (double) (this.rand.nextFloat() * 0.05F);
+                    entityitem.motionX += (double) ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.1F);
+                    entityitem.motionZ += (double) ((this.rand.nextFloat() - this.rand.nextFloat()) * 0.1F);
+                }
+            }
+        }
+        this.playSound(SoundEvents.ENTITY_SHEEP_SHEAR, 1.0F, 1.0F);
+    }
 
     public void writeAdditional(NBTTagCompound compound)
     {
