--- a/net/minecraft/entity/passive/EntityDolphin.java
+++ b/net/minecraft/entity/passive/EntityDolphin.java
@@ -3,6 +3,8 @@
 import java.util.List;
 import java.util.function.Predicate;
 import javax.annotation.Nullable;
+
+import carpet.helpers.entities.AbstractWaterCreature;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.IEntityLivingData;
@@ -56,7 +58,7 @@
 import net.minecraftforge.api.distmarker.Dist;
 import net.minecraftforge.api.distmarker.OnlyIn;
 
-public class EntityDolphin extends EntityWaterMob
+public class EntityDolphin extends AbstractWaterCreature
 {
     private static final DataParameter<BlockPos> TREASURE_POS = EntityDataManager.createKey(EntityDolphin.class, DataSerializers.BLOCK_POS);
     private static final DataParameter<Boolean> GOT_FISH = EntityDataManager.createKey(EntityDolphin.class, DataSerializers.BOOLEAN);
