--- a/net/minecraft/entity/EnumCreatureType.java
+++ b/net/minecraft/entity/EnumCreatureType.java
@@ -1,17 +1,16 @@
 package net.minecraft.entity;
 
+import carpet.helpers.entities.AbstractWaterCreature;
 import net.minecraft.entity.monster.IMob;
-import net.minecraft.entity.passive.EntityAmbientCreature;
-import net.minecraft.entity.passive.EntityAnimal;
-import net.minecraft.entity.passive.EntityWaterMob;
-import net.minecraft.entity.passive.IAnimal;
+import net.minecraft.entity.passive.*;
 
 public enum EnumCreatureType
 {
     MONSTER(IMob.class, 70, false, false),
     CREATURE(EntityAnimal.class, 10, true, true),
     AMBIENT(EntityAmbientCreature.class, 15, true, false),
-    WATER_CREATURE(EntityWaterMob.class, 15, true, false);
+    WATER_CREATURE(AbstractWaterCreature.class, 15, true, false),
+    WATER_AMBIENT(AbstractFish.class, 10, true, false);
 
     private final Class <? extends IAnimal > baseClass;
     private final int maxNumberOfCreature;
