--- a/net/minecraft/network/rcon/RConThreadMain.java
+++ b/net/minecraft/network/rcon/RConThreadMain.java
@@ -55,6 +55,12 @@
     {
         this.clientThreads = Maps.newHashMap();
     }
+    
+    // [CM] New method for rcon chat bridge
+    public void sendMessageCarpetChatBridge(String text)
+    {
+        this.clientThreads.forEach((s, r) -> r.sendMessageCarpetChatBridge(text));
+    }
 
     private void cleanClientThreadsMap()
     {
